fiftyone/core/map/batcher/id_batch.py

```diff
@@ -12,15 +12,15 @@
 import bson
 
 import fiftyone.core.utils as fou
-import fiftyone.core.map.batcher.batch as fomb
+import fiftyone.core.map.batcher.base as fomb
 from fiftyone.core.map.typing import SampleCollection
 
 fov = fou.lazy_import("fiftyone.core.view")
 
 T = TypeVar("T")
 
 
-class SampleIdBatch(fomb.SampleBatch):
+class SampleIdBatch(fomb.SampleBatcher, fomb.SampleBatch):
```

Comment:
Should this be SampleIdBatcher instead?

Reply:
No it's using interface segregation. I know there's no real interfaces in python but each implementation ("id" and "slice") are both the batch and batcher. This allows the other pieces that need them to specify only the batch or batcher interface.

Reply:
Removed the multi-inheritance to make it less confusing in the future.